resources/resourcebar.ts

```diff
@@ -1,96 +1,123 @@
   }
-  get bg() {
+  get bg(): string | null {
     return this.getAttribute('bg');
   }
 
-  set extraColor(c) {
-    this.setAttribute('extraColor', c);
+  set extraColor(c: string | null) {
+    this.setAttribute('extraColor', c ?? '');
   }
-  get extraColor() {
+  get extraColor(): string | null {
     return this.getAttribute('extraColor');
   }
 
   // Foreground color.
-  set fg(c) {
-    this.setAttribute('fg', c);
+  set fg(c: string | null) {
+    this.setAttribute('fg', c ?? '');
   }
-  get fg() {
+  get fg(): string | null {
     return this.getAttribute('fg');
   }
 
   // The width of the bar, in pixels (before |scale|).
-  set width(w) {
-    this.setAttribute('width', w);
+  set width(w: string | null) {
+    this.setAttribute('width', w ?? '');
   }
-  get width() {
+  get width(): string | null {
     return this.getAttribute('width');
   }
 
   // The height of the bar, in pixels (before |scale|).
-  set height(w) {
-    this.setAttribute('height', w);
+  set height(w: string | null) {
+    this.setAttribute('height', w ?? '');
   }
-  get height() {
+  get height(): string | null {
     return this.getAttribute('height');
   }
 
   // A value between 0 and |maxvalue|, indicating the amount of progress.
-  set value(s) {
-    this.setAttribute('value', s);
+  set value(s: string | null) {
+    this.setAttribute('value', s ?? '');
   }
-  get value() {
+  get value(): string | null {
     return this.getAttribute('value');
   }
 
   // A value between 0 and |maxvalue|, indicating the amount of "extra"
   // resource that exists.  Usually used for shields on a health bar.
-  set extraValue(s) {
-    this.setAttribute('extravalue', s);
+  set extraValue(s: string | null) {
+    this.setAttribute('extravalue', s ?? '');
   }
-  get extraValue() {
+  get extraValue(): string | null {
     return this.getAttribute('extravalue');
   }
 
   // The maximum value where when reached the progress bar will show 100%.
-  set maxvalue(s) {
-    this.setAttribute('maxvalue', s);
+  set maxvalue(s: string | null) {
+    this.setAttribute('maxvalue', s ?? '');
   }
-  get maxvalue() {
+  get maxvalue(): string | null {
     return this.getAttribute('maxvalue');
   }
 
   // If "right" then fills left-to-right (the default). If "left" then
   // fills right-to-left.
-  set toward(t) {
-    this.setAttribute('toward', t);
+  set toward(t: string | null) {
+    this.setAttribute('toward', t ?? '');
   }
-  get toward() {
+  get toward(): string | null {
     return this.getAttribute('toward');
   }
 
   // If "fill" then the progress goes empty-to-full, if "empty" then the
   // progress bar starts full and goes to empty.
-  set style(s) {
-    this.setAttribute('style', s);
+  set style(s: 'empty' | 'full' | null) {
+    this.setAttribute('style', s ?? 'empty');
   }
-  get style() {
-    return this.getAttribute('style');
+  get style(): 'empty' | 'full' | null {
+    return this.getAttribute('style') as 'empty' | 'full' | null;
```

Comment:
The `style` property here conflict the original implementation of `HTMLElement`, I think it should be changed to another name, like `styleFill` or something?

Reply:
Sounds good!